### PR TITLE
Bugfix/handle large policy and metadata collections

### PR DESCRIPTION
**Pull request by fieldju**

### CODE REVIEW

File: src/main/java/com/nike/cerberus/service/AuthenticationService.java [hunk 1]
```diff
@@ -247,6 +250,43 @@ private IamRoleAuthResponse authenticate(IamPrincipalCredentials credentials, Ma
...
+                originalPolicies = objectMapper.writeValueAsString(authResponse.getPolicies());
+            } catch (JsonProcessingException e) {
+                logger.warn("Failed to serialize original metadata or policies for token generated for IAM Principal: {}", credentials.getIamPrincipalArn());
+            }
+
+            authResponse.setMetadata(ImmutableMap.of("_truncated", "true"));
+            authResponse.setPolicies(ImmutableSet.of("_truncated"));
+
+            logger.error(
+                    "The auth token has length: {} which is > {} KMS cannot encrypt it, truncating token and removing policies and metadata " +
```
Comment (tlisonbee) [inline src/main/java/com/nike/cerberus/service/AuthenticationService.java]: Is this statement right?  You aren't truncating the token, are you?  You are truncating the policies and metadata.

File: src/main/java/com/nike/cerberus/service/AuthenticationService.java [hunk 2]
```diff
@@ -247,6 +250,43 @@ private IamRoleAuthResponse authenticate(IamPrincipalCredentials credentials, Ma
                     .withExceptionMessage("Failed to write IAM role authentication response as JSON for encrypting.")
                     .build();
         }
+
+        // if the metadata and policies make the token too big to encrypt with KMS we can as a stop gap trim the metadata and policies from the token
+        // This information is stored in Vault and can be fetched by the client with a look-up self call
+        if (authResponseJson.length > KMS_SIZE_LIMIT) {
+            String originalMetadata = "unknown";
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/service/AuthenticationService.java]: I think this is worth making a separate method and unit testing.

File: src/main/java/com/nike/cerberus/service/AuthenticationService.java (hunk 1 above)
Comment (fieldju) [inline src/main/java/com/nike/cerberus/service/AuthenticationService.java]: Yes, its truncated the auth payload, I will adjust the language.

File: src/main/java/com/nike/cerberus/service/AuthenticationService.java (hunk 2 above)
Comment (fieldju) [inline src/main/java/com/nike/cerberus/service/AuthenticationService.java]: I will refactor the new logic and write a unit test, thanks!

**Pull request merged**